### ISSUE TITLE
Implement Concept Exercise for booleans: Pac-Man Rules

### ISSUE THREAD

**Issue by kytrinyx** [exercism/haskell#1090]
We'd like to implement the Pac-Man Rules story to teach booleans in Haskell.
https://exercism.org/docs/building/tracks/stories/booleans-pac-man

The Elixir track implemented this exercise here:
https://github.com/exercism/elixir/tree/main/exercises/concept/pacman-rules
The exemplar solution is here:
https://github.com/exercism/elixir/blob/main/exercises/concept/pacman-rules/.meta/exemplar.ex

It should be possible to port this to Haskell without too much difficulty.

A suggested example solution, proposed by @MatthijsBlom is:

```
eatsGhost :: Bool -> Bool -> Bool
eatsGhost powerPelletActive touchingGhost =
    powerPelletActive && touchingGhost

scores :: Bool -> Bool -> Bool
scores touchingPowerPellet touchingDot =
    touching_power_pellet || touching_dot

loses :: Bool -> Bool -> Bool
loses powerPelletActive touchingGhost =
  not power_pellet_active && touching_ghost

wins :: Bool -> Bool -> Bool -> Bool
wins hasEatenAllDots powerPelletActive touchingGhost =
  hasEatenAllDots && not (loses powerPelletActive touchingGhost)
```

I suggest that we implement this exercise in two passes.

First a PR to build out the test, exemplar solution, and stub file.
We'll get feedback on this from a couple of people with a strong Haskell background, as well as having folks familiar with Exercism concept exercises review.

Once that is done, a second PR to write the instructions. For that one we'd like to get the instructions tested by someone who knows Haskell (but who hasn't been involved in working on the exercise). The way to test the instructions is to try to implement the exercise without referring to (or running) the tests.

Probably `Bool`, `||`, `&&`, `not`, and precedence will need to be explained in the introduction. We want to give the student the minimal amount of information required to solve this exercise, and for that they do need to know a tiny bit about operator precedence and how parentheses can be used to enforce this.

Pattern matching on `Bool`  is out of scope, as is _pointsfree style_.

Note that the supporting documentation for this exercise (the concept itself is currently in progress here: https://github.com/exercism/haskell/pull/1086

There's a draft PR that has some placeholder files here: https://github.com/exercism/haskell/pull/1088
Please feel free to grab all the files from there to make it as easy as possible to get started. Then we'll close that PR (no "real" work has been done there; I want whoever does the work to get the credit by opening their own PR)

**Comment by pwadsworth** [comment]
@kytrinyx I edited #1089 to reuse everything already built for Annalyn's Infiltration towards Pacman Rules.

**Comment by pwadsworth** [comment]
I intend to use the above example solution to create the exemplar, stub and tests tomorrow to complete "pass 1". 
I will check here before in case someone is already working on this.

**Comment by kytrinyx** [comment]
@pwadsworth it's all yours!

I'm so sorry about the mess with this. We're trying to figure out how to work on these concept exercises in the best possible way and I'm afraid you got caught in the middle of it.

I really do think that all of the stuff you've done so far on the booleans concept can be used for this! Hopefully once we have the code figured out the introduction etc will be easy to adapt from what you already worked on.

**Comment by ErikSchierboom** [comment]
@pwadsworth to make it even more clear, I've assigned you to this issue now :)

**Comment by pwadsworth** [comment]
I created draft PR #1093 with the exemplar, stub, hints and config files.
@MatthijsBlom is working on instructions.md I believe.

**Comment by pwadsworth** [comment]
This issue should be closed. I thought I'd added the "close #1090" in PR #1093.
